tests/openassetio-core-c/private/StringViewTest.cpp

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#include <string_view>
+#include <type_traits>
+
+#include <openassetio/c/StringView.h>
+
+#include <catch2/catch.hpp>
+
+#include <openassetio/typedefs.hpp>
+
+SCENARIO("Creating, modifying and querying a C API mutable StringView") {
+  GIVEN("A populated C++ string") {
+    openassetio::Str expectedStr;
+    constexpr std::size_t kExpectedCapacity = 123;
+    expectedStr.reserve(kExpectedCapacity);
+    expectedStr = "some string";
+
+    WHEN("a StringView is constructed wrapping the buffer") {
+      OPENASSETIO_NS(StringView)
+      actualStringView{expectedStr.capacity(), expectedStr.data(), expectedStr.size()};
+
+      THEN("StringView can be interrogated to reveal the values at construction") {
+        CHECK(actualStringView.maxSize == kExpectedCapacity);
+        CHECK(actualStringView.usedSize == expectedStr.size());
+        CHECK(std::string_view{actualStringView.buffer, actualStringView.usedSize} == expectedStr);
+      }
+
+      AND_WHEN("string is modified through the StringView") {
+        STATIC_REQUIRE(std::is_const_v<decltype(actualStringView.maxSize)>);
+        actualStringView.buffer[1] = '0';
+        actualStringView.usedSize = 4;
+
+        THEN("storage has been updated") { CHECK(expectedStr == "s0me string"); }
+
+        THEN("view has been updated") {
+          CHECK(std::string_view{actualStringView.buffer, actualStringView.usedSize} == "s0me");
+        }
+      }
```

Comment:
Updated tests for `StringView` in 37e01eb4 and docs in dd02b246, then docs and tests for `ConstStringView` in 26f408be. Split into multiple commits to try to make eventual squash easier.

Reply:
> Split into multiple commits to try to make eventual squash easier, sorry

Don't apologise, its a nightmare otherwise!

Reply:
[question] What are your thoughts on dealing with the length of the underlying buffer changing? Should we document that one should really generally be careful not to update the underlying buffer when it is exposed through a string view?

Reply:
I'll expand on the notes re. being careful with the underlying buffer, though it is C so we can hope people are grown-ups. 

Re. the length of the `data` buffer, I don't think that once set it can change, at least in "normal" C/C++, since this would require a reallocation, invalidating the `data` pointer (which is equally bad, if not worse).  

We could make the `capacity` non-const, so that it can be updated in case the user suddenly wants to use some of the bytes at the end of the `data` buffer for some other purpose, but that's getting pretty advanced.

One thing that just occurred to me to be aware of is that truncation in UTF-8 can mean a decoding failure.

Reply:
Added some more notes in daf0aaf6 and 0bd6b807

Also, somewhat tangentially but related, updated `StringView::data` to be a const pointer, so it can't be re-assigned once set, in 85f8ebf2

Reply:
> Re. the length of the data buffer, I don't think that once set it can change, at least in "normal" C/C++, since this would require a reallocation, invalidating the data pointer (which is equally bad, if not worse).

Sorry, was thinking of the `size` part, where you could change `"cat"` to `"ca"` in the underlying buffer, but `size` wouldn't be updated in the `StringView`.

Reply:
Simplified docstrings to simply say "don't use the buffer" in 0bac7408 and 8607eaf6, and removed the test that modifies the underlying buffer in 448599c2